### PR TITLE
Fix for Issue#532

### PR DESCRIPTION
**Pull request by daminichopra**
## Proposed changes

Included a metadata field in each entity table called `time_index_attribute` with type array(String). Stored the list of attributes in the field `time_index_attribute`. In the attribute related queries, retrieved rows where `time_index_attribute` includes the attribute or where `time_index_attribute` is `Null`.

## Types of changes

What types of changes does your code introduce to the project: _Put
an `x` in the boxes that apply_

- [ x ] Bugfix (non-breaking change which fixes an issue)
- [ x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
      functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after
creating the PR. If you're unsure about any of them, don't hesitate to
ask. We're here to help! This is simply a reminder of what we are going
to look for before merging your code._

- [ x ] I have read the [CONTRIBUTING][contrib] doc
- [ x ] I have signed the [CLA][cla]
- [ ] I have added tests that prove my fix is effective or that my
      feature works
- [ ] I have added necessary documentation (if appropriate)
- [ x ] Any dependent changes have been merged and published in
      downstream modules

## Further comments
Updated code changes for Fix of [Issue](https://github.com/orchestracities/ngsi-timeseries-api/issues/532)

[cla]: https://raw.githubusercontent.com/orchestracities/ngsi-timeseries-api/master/individual-cla.pdf
    "Martel Open Source Software Individual Contributor License Agreement"
[contrib]: https://github.com/smartsdk/ngsi-timeseries-api/blob/master/CONTRIBUTING.md
    "Contributing to QuantumLeap"

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -25,6 +25,7 @@ redis = "~=3.5"
 requests = ">=2.20"
 rq = "~=1.8"
 # run `pipenv install --dev` to get the packages below in your env
+pytest = "*"
```
Comment (chicco785) [inline Pipfile:28]: pytest is already included in the dev packages, to install them you need just to run `pipenv install --dev`

File: src/translators/sql_translator.py [hunk 2]
```diff
@@ -1411,11 +1427,7 @@ def _format_response(
                     elif original_name == self.TIME_INDEX_NAME:
                         v = self._get_isoformat(v)
                         if single_value:
-                            n = {
-                                'value': v,
-                                'type': 'DateTime'
-                            }
-                            e.setdefault('dateModified', n)
+                            e.setdefault('dateModified', v)
```
Comment (chicco785) [inline src/translators/sql_translator.py:1430]: i think you are using as base an old version of the code

File: src/translators/sql_translator.py [hunk 3]
```diff
@@ -527,10 +534,9 @@ def _preprocess_values(self, e, original_attrs, col_names,
 
     @staticmethod
     def _is_ngsi_array(attr, attr_t):
-        return (attr_t == NGSI_STRUCTURED_VALUE and 'value' in attr
-                and isinstance(attr['value'], list)) \
-            or ('value' in attr and isinstance(attr['value'], list)) \
-            or attr_t == "Array"
+        return (attr_t == NGSI_STRUCTURED_VALUE
+                and 'value' in attr and isinstance(attr['value'],
+                                                   list)) or attr_t == "Array"
```
Comment (chicco785) [inline src/translators/sql_translator.py:539]: i think you need to rebase using the proper master version

File: src/translators/sql_translator.py [hunk 4]
```diff
@@ -828,14 +834,23 @@ def _get_where_clause(self, entity_ids, from_date, to_date, fiware_sp=None,
                     + fiware_sp + "($|/.*)'")
         else:
             # Match prefix of fiware service path
-            clauses.append(" " + prefix + FIWARE_SERVICEPATH + " = ''")
+            clauses.append(" " + prefix + FIWARE_SERVICEPATH + " = '' ")
+        attrs = ''
+        attrs_clauses = []
+        attrs_clauses.append(" and ")
+        if attr_names is not None:
```
Comment (chicco785) [inline src/translators/sql_translator.py:841]: i am not sure the code added serve the purpose. the only place where you can compute the list of attributes that are linked to the time index, is where you compute the time_index. i.e.  `/src/reporter/timex.py`

File: src/translators/sql_translator.py (hunk 3 above)
Comment (daminichopra) [reply src/translators/sql_translator.py:539]: @chicco785 , I will fork new repo and then raise the PR for same.

File: src/translators/sql_translator.py (hunk 2 above)
Comment (daminichopra) [reply src/translators/sql_translator.py:1430]: > i think you are using as base an old version of the code

@chicco785  , I am working on this and I will update this PR.

**Pull request closed without merging**